pkg/webhook/v1beta1/pod/inject_webhook_test.go

```diff
@@ -651,32 +632,33 @@ func TestMutateMetricsCollectorVolume(t *testing.T) {
 		tc.SidecarContainerName,
 		tc.PrimaryContainerName,
 		tc.PathKind)
-	if err != nil {
-		t.Errorf("mutateMetricsCollectorVolume failed: %v", err)
-	} else if !equality.Semantic.DeepEqual(tc.Pod, tc.ExpectedPod) {
-		t.Errorf("Expected pod %v, got %v", tc.ExpectedPod, tc.Pod)
+	if diff := cmp.Diff(tc.WantError, err, cmpopts.EquateErrors()); len(diff) != 0 {
+		t.Errorf("Unexpected error from mutateMetricsCollectorVolume (-want,+got):\n%s", diff)
+	}
+	if err == nil && !equality.Semantic.DeepEqual(tc.Pod, tc.WantPod) {
+		t.Errorf("Unexpected error from mutateMetricsCollectorVolume, expected pod: %v, got: %v", tc.WantPod, tc.Pod)
```

Comment:
Could you apply the cmp approach here as well?

Reply:
Same as above.